lib/FusionInventory/Agent/Inventory.pm

```diff
@@ -250,7 +250,28 @@ sub addEntry {
         $entry->{SERIALNUMBER} = $entry->{SERIAL} if !$entry->{SERIALNUMBER}
     }
 
-    push @{$self->{content}{$section}}, $entry;
+    if ($params{identity}
+        && ref $params{identity} eq 'ARRAY'
```

Comment:
Well, my first comment here was lost in translation.
Please, don't use a list of callback, this is too complicated and dangerous for later use. Prefer to just use only one callback directly set as 'identity' parameter.
As I also commented `lib/FusionInventory/Agent/Task/Inventory/Generic/Dmidecode/Battery.pm`, I think callbacks could be merged or you need to explain why this is impossible.